### PR TITLE
[ticket/14129] Cache custom extensions autoloaders

### PR DESCRIPTION
**Pull request by rxu**
For phpBB 3.1 only. Based on #4142 for phpBB 3.2. Replaces #3885.
<a href="https://tracker.phpbb.com/browse/PHPBB3-14129">PHPBB3-14129</a>.

### CODE REVIEW

**Comment by Nicofuma** [comment]
@nickvergessen ok to have that in 3.1? It's kinda a new feature (a major perf improvement)

**Comment by Nicofuma** [comment]
@rxu you need to add the file to the cache driver (when purging)

**Comment by rxu** [comment]
@Nicofuma Oh indeed. Updated.

File: phpBB/phpbb/di/container_builder.php [hunk 1]
```diff
@@ -401,4 +415,45 @@ protected function get_container_filename()
...
+ */
+
+';
+		foreach ($installed_exts as $ext_name => $path)
+		{
+			// Load extension autoloaders
+			$filename = $path . 'vendor/autoload.php';
+			if (file_exists($filename))
+			{
+				$autoloaders .= "require '{$filename}';\n";
```
Comment (lavigor) [inline phpBB/phpbb/di/container_builder.php]: I guess parenthesis should be here too.

Comment (rxu) [inline phpBB/phpbb/di/container_builder.php]: Updated.

**Comment by marc1706** [comment]
Bump @nickvergessen

**Comment by marc1706** [comment]
:+1:

**Comment by nickvergessen** [comment]
I think we shouldn't do things like this in a maintenance release, the session-PR from 3.1.7 should be lesson enough...

**Pull request closed without merging**